generators/server/templates/src/main/java/package/config/_MicroserviceSecurityConfiguration.java

```diff
@@ -120,9 +128,26 @@ public TokenStore tokenStore() {
       @Bean
       public JwtAccessTokenConverter jwtAccessTokenConverter() {
           JwtAccessTokenConverter converter = new JwtAccessTokenConverter();
-          converter.setSigningKey(jHipsterProperties.getSecurity().getAuthentication().getJwt().getSecret());
-
+          converter.setVerifierKey(getKeyFromAuthorizationServer());
           return converter;
       }
+
+      @Bean
+      public RestTemplate loadBalancedRestTemplate(RestTemplateCustomizer customizer) {
+          RestTemplate restTemplate = new RestTemplate();
+          customizer.customize(restTemplate);
+          return restTemplate;
+      }
+
+      @Inject
+      @Qualifier("loadBalancedRestTemplate")
+      private RestTemplate keyUriRestTemplate;
+
+      private String getKeyFromAuthorizationServer() {
+          HttpEntity<Void> request = new HttpEntity<Void>(new HttpHeaders());
+          return (String) this.keyUriRestTemplate
```

Comment:
This is only part of the initial setup. Only happens when the app starts up...not every request. This is using the auth server's published public key so that each resource server doesn't need to hard-code it. Google publishes theirs here: https://www.googleapis.com/oauth2/v3/certs